### PR TITLE
Add dependabot to the repository

### PR DESCRIPTION
**Pull request by vdemeester**
# Changes

This might generate quite some pull-request initially and some of them
will always be red (bumping knative, kubernetes or tekton will
probably require some changes, like re-generate files, …).

But it should help to know when to update *and* will work with the
other type of dependencies.

Doing this based on
https://github.com/tektoncd/pipeline/pull/4915#issuecomment-1142232783 and https://github.com/tektoncd/cli/pull/1579.

Signed-off-by: Vincent Demeester <vdemeest@redhat.com>

/kind misc

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/operator/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    schedule:
+      interval: weekly
+
+  - package-ecosystem: "gomod" # See documentation for possible values
+    directory: "/" # Location of package manifests
+    schedule:
+      interval: "weekly"
+    labels:
+    - "ok-to-test"
+    - "dependencies"
```
Comment (piyush-garg) [inline .github/dependabot.yml:14]: should we add a label related to release note also

Comment (vdemeester) [reply .github/dependabot.yml:14]: good point, probably ?

Comment (piyush-garg) [reply .github/dependabot.yml:14]: Or may be we can customize the commit message can put the release note block there with some standard note regarding dependency

Comment (vdemeester) [reply .github/dependabot.yml:14]: > Or may be we can customize the commit message can put the release note block there with some standard note regarding dependency

I am not sure about that, but yeah that could be nice. *Or*, update our create-draft-release task to parse dependabot commits/PR description.

**Review by pradeepitm12 — APPROVED** [review]
/lgtm

**Comment by vdemeester** [comment]
Note: prior to https://github.com/tektoncd/operator/pull/833, it might be a bit *bumpy* 😝

**Review by savitaashture — COMMENTED** [review]
/lgtm 
:+1:

**Review by pradeepitm12 — COMMENTED** [review]
/lgtm

**Pull request merged**